src/tree_walker.js

```diff
@@ -47,6 +53,30 @@ TreeWalker.prototype.getCurrentParent = function() {
 
 
 /**
+ * @return {string} The current namespace to create Elements in.
+ */
+TreeWalker.prototype.getCurrentNamespace = function() {
+  return this.nsStack_[this.nsStack_.length - 1];
```

Comment:
I'd change this to 
Access at -1 may trigger dictionary lookup.
var len = this.nsStack_.length;
if (len === 0) {
  return undefined;
}
return this.nsStack_[len - 1];